### PR TITLE
Enforce apache thrift 0.9.3 in glide and in thrift generation

### PR DESCRIPTION
**Pull request by peter-edge**
This sets apache thrift to 0.9 in glide, and also makes sure we always use the released thrift in the tchannel-go releases for all generation.

https://github.com/uber/tchannel-go/issues/608

### CODE REVIEW

File: glide.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package: github.com/uber/tchannel-go
 import:
 - package: github.com/apache/thrift
-  version: master
+  version: ~0.9
```
Comment (prashantv) [inline glide.yaml]: should we do `^0.9.3`?

File: glide.lock [hunk 2]
```diff
@@ -1,33 +1,36 @@
-hash: 111bdf9797d23d4b9f8833a9780e382dd53153af70ad286f99815cea4c2fdf4b
-updated: 2016-08-05T18:02:17.171302481-04:00
+hash: 96546aa202f8964e8cc0c8acea5452cc9c6d79259ea4f9aaff03ffbf13c918a4
```
Comment (prashantv) [inline glide.lock]: Can we do the glide change in a separate PR -- since it upgrades a lot of other deps too. Let's make this change just affect the thrift compiler

File: glide.yaml (hunk 1 above)
Comment (breerly) [inline glide.yaml]: No, `~0.9` keeps us in the 0.9 minor.

Comment (prashantv) [inline glide.yaml]: I don't think we support `0.9.1` or `0.9.2` though in tchannel

**Comment by peter-edge** [comment]
What is the problem exactly? Do you want to lock to a micro?

On Thu, Mar 23, 2017 at 4:46 PM, Prashant Varanasi <notifications@github.com

File: glide.yaml (hunk 1 above)
Comment (peter-edge) [inline glide.yaml]: Note I don't think apache is going to be doing any updating, like I can set this to 0.9.3 if we don't support 0.9.1 or 0.9.2, but I don't think there will be a 0.9.4.

**Comment by breerly** [comment]
That depends: is the Thrift project releasing new functionality or breaking changes in patch releases (the z in x.y.z)?

If we can't trust them to do semver, then we might have to fully pin... that wouldn't be great tho.

**Comment by peter-edge** [comment]
I reverted that lock from this PR entirely.

My tl;dr:

1. Locking to either ~-0.9 or 0.9.3 is strictly an improvement over what we
have now, so IMO we should merge and merge and merge
2. In practice, it probably won't matter, as I don't think there will be
another 0.9 release
